conans/client/cmake.py

```diff
@@ -156,3 +165,48 @@ def runtime(self):
         if runtime:
             return "-DCONAN_LINK_RUNTIME=/%s" % runtime
         return ""
+
+    def configure(self, conan_file, args=None, vars=None, source_dir=None, build_dir=None):
+        args = args or []
+        vars = vars or {}
+        source_dir = source_dir or conan_file.conanfile_directory
+        self.build_dir = build_dir or self.build_dir or conan_file.conanfile_directory
+
+        mkdir(self.build_dir)
+        arg_list = _join_arguments([
+            self.command_line,
+            _args_to_string(args),
+            _vars_to_string(vars),
+            _args_to_string([source_dir])
+        ])
+        command = "cd %s && cmake %s" % (_args_to_string([self.build_dir]), arg_list)
+        conan_file.run(command)
+
+    def build(self, conan_file, args=None, build_dir=None, target=None):
+        args = args or []
+        build_dir = build_dir or self.build_dir or conan_file.conanfile_directory
+        if target is not None:
+            args = ["--target", target] + args
+
+        arg_list = _join_arguments([
+            _args_to_string([build_dir]),
+            self.build_config,
+            _args_to_string(args)
+        ])
+        command = "cmake --build %s" % arg_list
+        conan_file.run(command)
+
+
+def _vars_to_string(vars):
+    return _args_to_string('-D{0}={1}'.format(k, v) for k,v in vars.items())
+
+
+def _args_to_string(args):
```

Comment:
This function makes me a bit unconfortable, depending on an undocumented function of ``subprocess``. Also, I am not fully sure about the non win output with single quotes. So the output, as shown in your test:

```
'--foo \"bar\"' '-DSHARED=True' '/source'
```
I am a bit surprise, you don't have to write such single quotes in nix systems in the command line.
What am I missing here?